src/shutdown/shutdown.c

```diff
@@ -370,25 +391,8 @@ int main(int argc, char *argv[]) {
                    LOG_TARGET_KMSG))
                 bump_sysctl_printk_log_level(LOG_WARNING);
 
-        watchdog_device = getenv("WATCHDOG_DEVICE");
-        if (watchdog_device) {
-                r = watchdog_set_device(watchdog_device);
-                if (r < 0)
-                        log_warning_errno(r, "Failed to set watchdog device to %s, ignoring: %m",
-                                          watchdog_device);
-        }
-
-        watchdog_usec = getenv("WATCHDOG_USEC");
-        if (watchdog_usec) {
-                usec_t usec;
+        init_watchdog(getenv("WATCHDOG_DEVICE"), getenv("WATCHDOG_USEC"));
```

Comment:
i'd move the two getenv() calls into init_watchdog() as well, so that it takes no arguments, but encapsulates the logic comprehensively.

Reply:
Agreed.